### PR TITLE
Shuffles some tests around

### PR DESCRIPTION
**Pull request by kaapstorm**
These tests could probably use some more love, but shuffling them around a bit shaves off 5 seconds (from 30 to 25), and that's all I need to get a build to pass.

The diff is confusing. I just moved domain and user setup/teardown into class methods, and then I moved the tests that failed after that into a new test case called "ImporterTestsThatSetUpTheDomainEveryTime" :upside_down_face: 

The tests are identical except for one that now uses a context manager to clean up a user it creates.

### CODE REVIEW

File: corehq/apps/case_importer/tests/test_importer.py [hunk 1]
```diff
@@ -207,11 +321,39 @@ def testCaseLookupDomainCheck(self):
...
+            ['importer-test-external-id', 'age-1', 'sex-1', 'location-1'],
+            ['importer-test-external-id', 'age-2', 'sex-2', 'location-2'],
+        )
+
+        # the first one should create the case, and the remaining two should update it
+        res = do_import(file, config, self.domain)
+        self.assertEqual(1, res['created_count'])
+        self.assertEqual(2, res['match_count'])
+        self.assertFalse(res['errors'])
+        self.assertEqual(2, res['num_chunks']) # the lookup causes an extra chunk
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/apps/case_importer/tests/test_importer.py:348]: Ignoring

**Review by millerdev — APPROVED** [review]
> shuffling them around a bit shaves off 5 seconds (from 30 to 25), and that's all I need to get a build to pass

Do you think the run time variability is low enough that 5s of headroom will not result in intermittent failures?

**Comment by kaapstorm** [comment]
> 5s of headroom will not result in intermittent failures?

This is an old test suite. Manish made a few changes recently, and I think that's what pushed it over the edge. But I think a bigger cause of this test timing out is the ES-related issues that #27849 addresses. I'm quietly waiting for that PR to be merged, and then I'll see whether we still need this one.

**Comment by kaapstorm** [comment]
Looks like these changes are no longer necessary.

**Pull request closed without merging**